faq/mpi-legacy.inc

```diff
@@ -0,0 +1,593 @@
+of the function: [MPI_Errhandler_create()] $right_arrow
+[MPI_Comm_create_errhandler()].
+
+_Technically_, the type of the first parameter also changed
+( [MPI_Handler_function] $right_arrow [MPI_Comm_errhandler_function]),
+but most applications do not use this type directly and may not even
+notice the change.
+
+<geshi c>
+void my_errhandler_function(MPI_Comm *comm, int *code, ...)
+{
+    // Do something useful to handle the error
+}
+
+void some_function(void)
+{
+    MPI_Errhandler my_handler;
+
+    // Old way
+    MPI_Errhandler_create(my_errhandler_function, &my_handler);
+
+    // New way
+    MPI_Comm_create_errhandler(my_errhandler_function, &my_handler);
+}
+</geshi>
+
+In Fortran, only the subroutine name changed: [MPI_ERRHANDLER_CREATE]
+$right_arrow [MPI_COMM_CREATE_ERRHANDLER].
+
+<geshi fortran>
+ USE mpi
+ EXTERNAL my_errhandler_function
+ INTEGER ierror
+ INTEGER my_handler
+
+ ! Old way
+ CALL MPI_ERRHANDLER_CREATE(my_errhandler_function, my_handler, ierror)
+
+ ! Old way
+ CALL MPI_COMM_CREATE_ERRHANDLER(my_errhandler_function, my_handler, ierror)
+</geshi>";
+
+/////////////////////////////////////////////////////////////////////////
+
+$q[] = "How do I update my MPI application to stop using MPI_ERRHANDLER_GET?";
+
+$anchor[] = "mpi-1-mpi-errhandler-get";
+
+$a[] = "In both C and Fortran, the only thing that changed with
+regards to [MPI_ERRHANDLER_GET] is the name: [MPI_ERRHANDLER_GET]
+$right_arrow [MPI_COMM_GET_ERRHANDLER].
+
+All parameter types stayed the same.";
+
+/////////////////////////////////////////////////////////////////////////
+
+$q[] = "How do I update my MPI application to stop using MPI_ERRHANDLER_SET?";
+
+$anchor[] = "mpi-1-mpi-errhandler-set";
+
+$a[] = "In both C and Fortran, the only thing that changed with
+regards to [MPI_ERRHANDLER_SET] is the name: [MPI_ERRHANDLER_SET]
+$right_arrow [MPI_COMM_SET_ERRHANDLER].
+
+All parameter types stayed the same.";
+
+/////////////////////////////////////////////////////////////////////////
+
+$q[] = "How do I update my MPI application to stop using MPI_TYPE_HINDEXED?";
+
+$anchor[] = "mpi-1-mpi-type-hindexed";
+
+$a[] = "In both C and Fortran, _effectively_ the only change is the
+name of the function: [MPI_TYPE_HINDEXED] $right_arrow
+[MPI_TYPE_CREATE_HINDEXED].
+
+In C, the new function also has a [const] attribute on the two array
+parameters, but most applications won't notice the difference.
+
+All other parameter types stayed the same.
+
+<geshi c>
+int count = 2;
+int block_lengths[] = { 1, 2 };
+MPI_Aint displacements[] = { 0, sizeof(int) };
+MPI_Datatype newtype;
+
+// Old way
+MPI_Type_hindexed(count, block_lengths, displacements, MPI_INT, &newtype);
+
+// New way
+MPI_Type_create_hindexed(count, block_lengths, displacements, MPI_INT, &newtype);
+</geshi>";
+
+/////////////////////////////////////////////////////////////////////////
+
+$q[] = "How do I update my MPI application to stop using MPI_TYPE_HVECTOR?";
```

Comment:
Why does TYPE_HVECTOR not get an example?

Reply:
It was just like `MPI_ERRHANDLER_GET|SET` -- they didn't get examples, either, because literally just the name changed; nothing else.

But I don't feel strongly -- do you want examples for all 3 of these?